client/components/domains/registrant-extra-info/ca-form.jsx

```diff
@@ -0,0 +1,152 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import {
+	camelCase,
+	difference,
+	isEmpty,
+	keys,
+	map,
+	pick,
+} from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getContactDetailsExtraCache } from 'state/selectors';
+import { updateContactDetailsCache } from 'state/domains/management/actions';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSelect from 'components/forms/form-select';
+import FormCheckbox from 'components/forms/form-checkbox';
+
+const ciraAgreementUrl = 'https://services.cira.ca/agree/agreement/agreementVersion2.0.jsp';
+const defaultValues = {
+	legalType: 'CCT',
+	ciraAgreementAccepted: false,
+};
+
+class RegistrantExtraInfoCaForm extends React.PureComponent {
+	static propTypes = {
+		contactDetailsExtra: PropTypes.object.isRequired,
+		translate: PropTypes.func.isRequired,
+	}
+
+	constructor( props ) {
+		super( props );
+		const { translate } = props;
+		const legalTypes = {
+			ABO: translate( 'Aboriginal' ),
+			ASS: translate( 'Association (Unincorporated)' ),
+			CCO: translate( 'Canadian Corporation' ),
+			CCT: translate( 'Canadian Citizen' ),
+			EDU: translate( 'Educational Institution' ),
+			GOV: translate( 'Government' ),
+			HOP: translate( 'Hospital' ),
+			INB: translate( 'Indian Band' ),
+			LAM: translate( 'Library, Archive, or Museum' ),
+			LGR: translate( 'Legal Representative' ),
+			MAJ: translate( 'Her Majesty the Queen' ),
+			// An official mark is Canadian thing like a trademark for public authorities
```

Comment:
I think this should be a translator comment. Maybe actually add a context to all of them `context: 'Canadian entity type'`. Not sure, but it shouldn't stay this way.